### PR TITLE
[AME] Seperate management API configs for Gallery and Search Service

### PR DESCRIPTION
**Pull request by zhhyu**
Provide the flexibility for the AzureManagementAPIWrapperConfiguration from different tenants.

### CODE REVIEW

File: src/NuGet.Services.EndToEnd/Support/Clients/Clients.cs [hunk 1]
```diff
@@ -94,8 +93,34 @@ private static Clients InitializeInternal(TestSettings testSettings)
                 symbolServerClient);
         }
 
-        private static IRetryingAzureManagementAPIWrapper GetAzureManagementAPIWrapper(TestSettings testSettings)
+        private static IRetryingAzureManagementAPIWrapper GetAzureManagementAPIWrapperForGallery(TestSettings testSettings)
         {
+            if (testSettings.GalleryConfiguration.AzureManagementAPIWrapperConfiguration != null)
```
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/Support/Clients/Clients.cs]: these methods could be simplified if you have one helper that took a `AzureManagementAPIWrapperConfiguration` and returned a `RetryingAzureManagementAPIWrapper` then these two methods call into that helper twice.

File: src/NuGet.Services.EndToEnd/Support/Clients/Clients.cs [hunk 2]
```diff
@@ -73,12 +73,11 @@ public static Clients Initialize(TestSettings testSettings)
         /// </summary>
         private static Clients InitializeInternal(TestSettings testSettings)
         {
-            var azureManagementAPI = GetAzureManagementAPIWrapper(testSettings);
-            
+
```
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/Support/Clients/Clients.cs]: nit: no extra line

**Review by joelverhagen — APPROVED** [review]
Nice one!

File: src/NuGet.Services.EndToEnd/Support/Clients/Clients.cs (hunk 1 above)
Comment (zhhyu) [reply src/NuGet.Services.EndToEnd/Support/Clients/Clients.cs]: Decide to keep the original one.

**Comment by zhhyu** [comment]
Tests successfully:
58084, 58086, 58087.

**Pull request merged**